modules/mapper-extras/src/main/java/org/elasticsearch/index/mapper/extras/MatchOnlyTextFieldMapper.java

```diff
@@ -246,6 +247,11 @@ public Query termQuery(Object value, SearchExecutionContext context) {
             return new ConstantScoreQuery(super.termQuery(value, context));
         }
 
+        @Override
+        public Query exactQuery(Object value, SearchExecutionContext context) {
+            return new TextFieldExactQuery(this, context.getForField(this, FielddataOperation.SCRIPT), value.toString());
```

Comment:
I'm wondering if this should be `SOURCE` instead of `SCRIPT` for two reasons:
1. To me it makes it more obvious where this field is trying to get it's field data from.
2. Should the scripting behavior ever need to change, this could be easily missed as a side effect of that change.

I realize this would add a bit of additional logic to `fielddataBuilder`, but I think the clarity may be a good trade off here.